File: stat-map.js
#!/usr/bin/nodejs

var readline = require('readline');
var count = parseInt(process.env.FILE_COUNT);

var rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout,
	terminal: false
});

rl.on('line', function (line) {
	var values = line.split('\t'), term = values[0], file = values[1], tf = parseFloat(values[2]), length = parseFloat(values[3]), df = parseFloat(values[4]);
	console.log([term, file, tf / length * Math.log(count / df)].join('\t'));
});

File: stat-map.py
from __future__ import division
import fileinput
import math
import os

count = int(os.environ['FILE_COUNT'])

for line in fileinput.input():
  term, file, tf, length, df = line.split()
  print("%s\t%s\t%f" % (term, file, int(tf) / int(length) * math.log(count / int(df))))